### PR TITLE
Interaction tests for combobox & add data test ids by prop to clickable, search, textfield, combobox

### PR DESCRIPTION
**Pull request by hadasfa**

### CODE REVIEW

File: src/components/Combobox/__tests__/combobox-interactions.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  clickElement,
+  resetFocus,
+  typeText,
+  interactionSuite,
+  pressNavigationKey
+} from "../../../__tests__/interactions-helper";
+import { queryByText } from "@storybook/testing-library";
+import { getTestId, NAVIGATIONS_COMMANDS } from "utils/test-utils";
+
+async function getSearchElement(canvas) {
```
Comment (laviomri) [inline src/components/Combobox/__tests__/combobox-interactions.js:16]: I think the function name needs to be updated (you don't get just the search element). WDYT about `getComponentElements`?

File: src/components/Combobox/__tests__/combobox-interactions.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  const { comboboxElement, searchElement } = await getSearchElement(canvas);
+  await typeText(searchElement, "jjj");
+  const cleanSearchButton = getByTestId(
+    comboboxElement,
+    getTestId(ELEMENT_TYPES.CLEAN_SEARCH_BUTTON, "combobox-search")
+  );
+  await clickElement(cleanSearchButton);
+  expect(searchElement).toHaveValue("");
+  const option1 = getByText(comboboxElement, "Option 1");
+  expect(option1).toBeTruthy();
```
Comment (laviomri) [inline src/components/Combobox/__tests__/combobox-interactions.js:38]: WDYT about
```suggestion
  expect(option1).toBeInTheDocument();
```
or:
```
expect(option1).toBeVisible();
```

I think it's more "accurate"

File: src/components/Combobox/__tests__/combobox-interactions.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+// there is no story which is not overview which activate this ability
+// eslint-disable-next-line no-unused-vars
```
Comment (laviomri) [inline src/components/Combobox/__tests__/combobox-interactions.js:42]: I'm not sure I understand - is the function unused? why?

File: src/components/TextField/TextField.jsx [hunk 4]
```diff
@@ -143,6 +146,7 @@ const TextField = forwardRef(
               aria-owns={searchResultsContainerId}
               aria-activedescendant={activeDescendant}
               required={required}
+              data-testid={dataTestId ? dataTestId : getTestId(ELEMENT_TYPES.TEXT_FIELD, id)}
```
Comment (laviomri) [inline src/components/TextField/TextField.jsx:149]:
```suggestion
              data-testid={dataTestId || getTestId(ELEMENT_TYPES.TEXT_FIELD, id)}
```

File: src/components/TextField/TextField.jsx [hunk 5]
```diff
@@ -181,6 +185,7 @@ const TextField = forwardRef(
               })}
               onClick={onIconClickCallback}
               tabIndex={!shouldFocusOnSecondaryIcon ? "-1" : "0"}
+              dataTestId={secondaryDataTestId ? secondaryDataTestId : getTestId(ELEMENT_TYPES.TEXT_FIELD_SECONDARY_BUTTON, id)}
```
Comment (laviomri) [inline src/components/TextField/TextField.jsx:188]:
```suggestion
              dataTestId={secondaryDataTestId || getTestId(ELEMENT_TYPES.TEXT_FIELD_SECONDARY_BUTTON, id)}
```

File: src/components/VirtualizedList/__stories__/VirtualizedList.stories.mdx [hunk 6]
```diff
@@ -2,13 +2,47 @@ import VirtualizedList from "../VirtualizedList";
...
 <!--- Component template -->
 
-export const virtualizedListTemplate = createComponentTemplate(VirtualizedList);
+export const virtualizedListTemplate = (args) => {
+    const itemRenderer = useCallback((item, index, style) => {
+        const backgroundColor = index % 2 === 0 ? "white" : "#f8f8f0";
+        return (
+            <div key={index} style={style}>
+                <div
+                    style={{
```
Comment (laviomri) [inline src/components/VirtualizedList/__stories__/VirtualizedList.stories.mdx:19]: consider extracting styles to scss?

**Review by laviomri — APPROVED** [review]
cool! had some comments

File: src/components/VirtualizedList/__stories__/VirtualizedList.stories.mdx (hunk 6 above)
Comment (hadasfa) [reply src/components/VirtualizedList/__stories__/VirtualizedList.stories.mdx:19]: can't be done. this is a dynamic css which position the visible elements in the virtualized list in the right position according to the scroll.

Comment (laviomri) [reply src/components/VirtualizedList/__stories__/VirtualizedList.stories.mdx:19]: I see it now. if you want, you can move all the other rules to a css class, and keep only the width here (like you probably would have done in a "real" component). but it's not critical :)

**Pull request merged**